ccmlib/scylla_repository.py

```diff
@@ -0,0 +1,127 @@
+from __future__ import with_statement
+
+import os
+import tarfile
+import tempfile
+import time
+import shutil
+import glob
+
+from six import print_
+from six.moves import urllib
+
+from ccmlib.common import (ArgumentError, CCMError, get_default_path, rmdirs, validate_install_dir)
+from ccmlib.repository import __download
+
+GIT_REPO = "http://github.com/scylladb/scylla.git"
+
+RELOCATABLE_URLS_BASE = 'https://s3.amazonaws.com/downloads.scylladb.com/relocatable/{0}/{1}'
+
+
+def setup(version, verbose=True):
+    s3_url = 'https://s3.amazonaws.com/downloads.scylladb.com/relocatable/unstable/master/latest'
+    type_n_version = version.split(':')
+    if len(type_n_version) == 2:
+        s3_version = type_n_version[1]
+        s3_url = RELOCATABLE_URLS_BASE.format(type_n_version[0], s3_version)
+        version = os.path.join(*type_n_version)
+
+    cdir = version_directory(version)
```

Comment:
if we failed to download some part - the repo will be corrupt and the user will be stuck - better make sure we protect against this - e.g. temp direcoty and rename or something in those lines

Reply:
I've address this, and now the download and untar is done into a temp directory and the moved into place once everything succeeded.